ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -248,12 +251,14 @@ public org.omg.CosTransactions.Vote prepare() throws HeuristicMixed,
                 case XAException.XA_RBOTHER:
                 case XAException.XA_RBPROTO:
                 case XAException.XA_RBTIMEOUT:
-                    // No txn -> assume rollback
+                    if (associationEnded) {
+                        // No txn -> assume rollback
 
-                    // Rolled back by specs
-                    // This will avoid calling rollback when aborting
-                    _rolledBack = true;
-                    return Vote.VoteRollback;
+                        // Rolled back by specs
+                        // This will avoid calling rollback when aborting
+                        _rolledBack = true;
+                        return Vote.VoteRollback;
```

Comment:
Shouldn't the `return` instruction be outside the `if`?

Reply:
Just to note I have seen this and for now I consider it as I did above on the local JTA case so will leave the conversation on there if possible but +1 to having this thread for tracking.